services/server/src/handlers/availabilitySlot.ts

```diff
@@ -0,0 +1,235 @@
+          })),
+        ],
+      },
+    ],
+    total: paginatedSlots.total,
+  };
+
+  res.status(200).json(result);
+}
+
+/**
+ * This single endpoint is dedicated to create, update, and delete slots
+ * This is because when the user is preparing his schedule, he can create,
+ * delete, and update simultaneously.
+ */
+async function set(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isTutor(user) || isTutorManager(user);
+  if (!allowed) return next(forbidden());
+
+  const payload = setPayload.parse(req.body);
+
+  const creates = payload.slots.filter(
+    (obj) => obj.action === "create"
+  ) as Array<CreateAction>;
+  const updates = payload.slots.filter(
+    (obj) => obj.action === "update"
+  ) as Array<UpdateAction>;
+  const deletes = payload.slots.filter(
+    (obj) => obj.action === "delete"
+  ) as Array<DeleteAction>;
+
+  const error = await knex.transaction(async (tx) => {
+    const deleteIds = deletes.map((obj) => obj.id);
+    const ids = [...deleteIds, ...updates.map((obj) => obj.id)];
+
+    if (!isEmpty(ids)) {
+      const allExist = await availabilitySlots.allExist(ids, tx);
+      if (!allExist) return notfound.slot();
+
+      const isOwner = await availabilitySlots.isOwner({
+        slots: ids,
+        owner: user.id,
+        tx,
+      });
+      if (!isOwner) return forbidden();
+
+      // delete slots
+      if (!isEmpty(deletes)) {
+        const associatedLessons = await lessons.find({ slots: deleteIds, tx });
+        const associatedInterviews = await interviews.find({
+          slots: deleteIds,
+          tx,
+        });
+        const associatesCount =
+          associatedLessons.total + associatedInterviews.total;
+        if (associatesCount > 0) {
+          await lessons.cancelBatch({
+            ids: associatedLessons.list.map((l) => l.id),
+            canceledBy: user.id,
+            tx,
+          });
+          await interviews.cancelBatch({
+            ids: associatedInterviews.list.map((i) => i.ids.self),
+            canceledBy: user.id,
+            tx,
+          });
+          await availabilitySlots.markAsDeleted({ ids: deleteIds, tx });
+        } else {
+          await availabilitySlots.delete(deleteIds, tx);
+        }
+      }
+
+      // update slots
+      if (!isEmpty(updates))
+        for (const update of updates) {
+          await availabilitySlots.update(
+            update.id,
+            {
+              start: update.start,
+              end: update.end,
+            },
+            tx
+          );
+        }
+    }
+
+    if (!isEmpty(creates)) {
+      // check for confliction in creates list
+      const mySlots = await availabilitySlots.find({ users: [user.id] });
+      for (const create of creates) {
+        if (
+          isIntersecting(
+            { id: 0, start: create.start, end: create.end },
+            mySlots.list
+          )
+        )
+          return next(conflict());
+      }
```

Comment:
We can make it more clean by doing the changes below: 


```suggestion
    const intersecting = creates.find(create => 
        isIntersecting(
        { id: 0, start: create.start, end: create.end },
        mySlots.list
        )
    );
    if (intersecting) return next(conflict());
```